pinot-core/src/test/java/com/linkedin/pinot/core/segment/store/SegmentDirectoryPathsTest.java

```diff
@@ -34,4 +38,50 @@ public void testSegmentPathFor() {
     Assert.assertEquals(v2Dir, f);
     Assert.assertEquals(v3Dir, new File("f", SegmentDirectoryPaths.V3_SUBDIRECTORY_NAME));
   }
+
+  @Test
+  public void testFindMetadataFile()
+      throws IOException {
+    File tempDirectory = null;
+    try {
+      // setup temp dir, v3 subdir and create metadata.properties in both
+      tempDirectory = new File(SegmentDirectoryPaths.class.toString());
+      FileUtils.forceMkdir(tempDirectory);
```

Comment:
Do you want to make this as deleteOnExit() as well? I know the file is deleted in the finally block, but just in case the process dies....